src/MSALWrapper.Test/AuthFlow/AuthFlowFactoryTest.cs

```diff
@@ -135,9 +145,45 @@ public void Windows_Defaults()
 
             IEnumerable<IAuthFlow> subject = this.Subject(AuthMode.Default);
 
-            subject.Should().HaveCount(1);
+            subject.Should().HaveCount(2);
+            var names = subject.Select(a => a.GetType().Name).ToList();
+            names[0].Should().Be(typeof(IntegratedWindowsAuthentication).Name);
+            names[1].Should().Be(typeof(Web).Name);
+        }
+
+        [Test]
+        public void Windows10Or11_All()
+        {
+            this.MockIsWindows10Or11(true);
+
+            IEnumerable<IAuthFlow> subject = this.Subject(AuthMode.All);
+
+            subject.Should().HaveCount(4);
+
+            // BeEquivalentTo doesn't assert order for a list :(
+            // so explicitly assert the first and second item names.
+            var names = subject.Select(a => a.GetType().Name).ToList();
+            names[0].Should().Be(typeof(IntegratedWindowsAuthentication).Name);
+            names[1].Should().Be(typeof(Broker).Name);
+            names[2].Should().Be(typeof(Web).Name);
+            names[3].Should().Be(typeof(DeviceCode).Name);
+        }
+
+        [Test]
+        public void Windows_All()
+        {
+            this.MockIsWindows10Or11(false);
+
+            IEnumerable<IAuthFlow> subject = this.Subject(AuthMode.All);
+
+            subject.Should().HaveCount(3);
+
+            // BeEquivalentTo doesn't assert order for a list :(
+            // so explicitly assert the first and second item names.
             var names = subject.Select(a => a.GetType().Name).ToList();
-            names[0].Should().Be(typeof(Web).Name);

```

Comment:
Shouldn't `windows_all` also have broker mode?

Reply:
Broker only works for Windows 10 and 11. If you see we have a test that checks flows that will queue up if windows version is 10 or 11.